src/phoenix/server/app.py

```diff
@@ -660,6 +683,14 @@ def create_app(
     )
     last_updated_at = LastUpdatedAt()
     middlewares: List[Middleware] = [Middleware(HeadersMiddleware)]
+    if origins := get_env_csrf_trusted_origins():
+        trusted_hostnames = [h for o in origins if o and (h := urlparse(o).hostname)]
+        middlewares.append(Middleware(RequestOriginHostnameValidator, trusted_hostnames))
+    elif email_sender or oauth2_client_configs:
+        logger.warning(
+            "CSRF protection is disabled because trusted origins have not been set via "
+            f"the `{ENV_PHOENIX_CSRF_TRUSTED_ORIGINS}` environment variable."
+        )
```

Comment:
```suggestion
    else:
        logger.warning(
            "CSRF protection is disabled because trusted origins have not been set via "
            f"the `{ENV_PHOENIX_CSRF_TRUSTED_ORIGINS}` environment variable."
        )
```

Do we want to warn if CSRF protection is not set, even if no OAuth2 IDP or SMTP relay is configured? Also makes sense to me to not warn since this feature is opt-in.

Reply:
not sure what's better. let's punt and leave it as is since it's less intrusive

Reply:
Okay. Also okay to nix altogether, I think.

Reply:
I think it's good to raise some awareness: it's a middle ground in terms of intrusiveness.